### ISSUE TITLE
Regression in depascalize in 1.3.0?

### ISSUE THREAD

**Issue by mm-matthias** [nficano/humps#16]
I want to convert strings to snake_case.

The following used to work with pyhumps 1.2.2:

```python
>>> humps.depascalize('error_400')
... 'error_400'
>>> humps.depascalize('error_a32')
... 'error_a32'
```

Pyhumps 1.3.0 gives:

```python
>>> humps.depascalize('error_400')
... 'error__400'
>>> humps.depascalize('error_a32')
... 'error_a_32'
```

The results returned by 1.3.0 look wrong to me. The documentation of `depascalize` says `Convert a string, dict, or list of dicts to snake case.`. The original strings are already snake case, so nothing should happen. This only seems to happen if there's a sequence of underscore followed by number. If you convert a string without numbers 1.3.0 returns the right answer:

```python
>>> humps.depascalize('error_a')
... 'error_a'
```

What do you think @nficano ?

**Comment by nficano** [comment]
I think I'll revert until I can sort it out. thanks for the heads up!

**Comment by mm-matthias** [comment]
Thanks for the quick reaction. The problem is not fixed with 1.3.1 though as this release only seems to revert the "depth" parameter changes. The issue seems to have appeared with this commit: https://github.com/nficano/humps/commit/fa93c3172841ea048b5b9ce07356ea68ec12983a . When I revert the change to the regex I get the expected results.
